libexec/pyenv-version-file-read

```diff
@@ -5,18 +5,17 @@ set -e
 
 VERSION_FILE="$1"
 
-if [ -e "$VERSION_FILE" ]; then
+if [ -s "$VERSION_FILE" ]; then
   # Read the first non-whitespace word from the specified version file.
   # Be careful not to load it whole in case there's something crazy in it.
   IFS="${IFS}"$'\r'
-  words=($(cut -b 1-1024 "$VERSION_FILE" | sed -n 's/^[[:space:]]*\([^[:space:]#][^[:space:]]*\).*/\1/p'))
-  versions=("${words[@]}")
-
-  if [ -n "$versions" ]; then
-    IFS=":"
-    echo "${versions[*]}"
-    exit
-  fi
+  sep=
+  while read -n 1024 -r version _; do
+      [[ -z $version || $version == \#* ]] && continue
+      printf "%s%s" "$sep" "$version"
```

Comment:
You never print a newline at the end?

Reply:
This script is only executed and its output captured so that trailing newlines, if any, are discarded anyway, so it doesn't make a practical difference. It wouldn't hurt either, though, if that's preferred.

Reply:
This file is only ever read in contexts that discard trailing newlines, if any, anyway, so it makes no practical difference here. We could add one if that's preferred though.

Reply:
Added in the revised commit 569718495b79d6073e37619866a015d961145f34.